### PR TITLE
HAWQ-734. Refactor feature test build and some bugfix

### PR DESCRIPTION
**Pull request by xunzhang**

### CODE REVIEW

**Comment by xunzhang** [comment]
cc @wengyanqing @ztao1987 @zhangh43

**Comment by xunzhang** [comment]
cc @huor

File: src/test/feature/Makefile [hunk 1]
```diff
@@ -1,32 +1,28 @@
...
 override CXXFLAGS = -Wall -O0 -g -std=c++11
-override CPPFLAGS := -I/usr/include -I/usr/local/include -I/usr/include/libxml2 -I$(top_builddir)/src/test/feature/ -I$(top_builddir)/src/test/feature/lib/ -I$(top_builddir)/src/interfaces/libpq -I$(top_builddir)/src/interfaces -I$(top_builddir)/src/include  -I$(gtest_include) -I$(gmock_include)
-
+override CPPFLAGS := -I/usr/include -I/usr/local/include -I/usr/include/libxml2 -I$(abs_top_srcdir)/src/test/feature/ -I$(abs_top_srcdir)/src/test/feature/lib/ -I$(abs_top_srcdir)/src/interfaces/libpq -I$(abs_top_srcdir)/src/interfaces -I$(abs_top_srcdir)/src/include  -I$(gtest_include) -I$(gmock_include)
 override LIBS := $(LIBS) -lgtest -lpq -lxml2 -ltest
-override LDFLAGS += -L/usr/local/lib -L/usr/lib -L$(top_builddir)/src/test/feature/lib -L$(gtest_lib_path) -L$(gmock_lib_path)
+override LDFLAGS += -L/usr/local/lib -L/usr/lib -L$(abs_top_srcdir)/src/test/feature/lib -L$(gtest_lib_path) -L$(gmock_lib_path)
 
-PROG = test-main.cpp testlib/test-lib.cpp
+PROG = test-main.cpp $(wildcard testlib/*.cpp)
```
Comment (huor) [inline src/test/feature/Makefile]: It is possible to make the wildcard to include all test folders? i.e., "+PROG = test-main.cpp $(wildcard */*.cpp)"

**Comment by wengyanqing** [comment]
+1

File: src/test/feature/lib/sql-util.cpp [hunk 2]
```diff
@@ -79,7 +79,7 @@ void SQLUtility::execSQLFile(const std::string &sqlFile,
 std::unique_ptr<PSQL> SQLUtility::getConnection() {
   std::unique_ptr<PSQL> psql(
       new PSQL(HAWQ_DB, HAWQ_HOST, HAWQ_PORT, HAWQ_USER, HAWQ_PASSWORD));
-  return std::move(psql);
+  return psql;
```
Comment (ztao1987) [inline src/test/feature/lib/sql-util.cpp]: Here I returned pointer ownership, otherwise this memory will be deleted by unique pointer

**Comment by ztao1987** [comment]
+1

File: src/test/feature/lib/sql-util.cpp (hunk 2 above)
Comment (xunzhang) [inline src/test/feature/lib/sql-util.cpp]: ok, modified back.

**Pull request merged**